### ISSUE TITLE
Able to have matching but if else statement overriding

### ISSUE THREAD

**Issue by pdavies88** [ga-dc/project1#110]
I am able to match my images and clear the array and reset the counter to then click on new images to try to match. However if I click on 2 images that are not a match the image and class are over ridden for all elements not just the two I selected. Trying to figure out how to keep the matched images the same while having the non matched elements revert to their original state without overriding the whole board that includes the matched elements. Below is my code any advice is appreciated:

``````
$(document).ready(function(){
 var checkingArray = [];
    var numFlipped = 0;
    $('div div').on("click",function(){
      if (!$(this).hasClass('flip')){
        $(this).addClass('flip')
        pickedCard = $(this).find("img.frontCard");
        CardValue = $(this).val();
        console.log(CardValue);
        pickedCard.attr("src", imageLibrary[CardValue])
        checkingArray.push(CardValue)
        console.log(checkingArray);
        numFlipped++;
        console.log(numFlipped)
      }
      else {
        console.log("already picked!")
      }
      cardMatching();
      cardNotMatching();
    })


function cardNotMatching(){
  if (numFlipped === 2){
    if(checkingArray[0] !== checkingArray[1]) {
      setTimeout (
      function(){
      numFlipped = 0;
      checkingArray = [];
      $("div div").removeClass("flip");
      $("img.frontCard").attr("src", "images/Mortal_Kombat_Logo.png")
    },
    1500);
    }
  }
}
function cardMatching(){
  if (numFlipped === 2){
    if(checkingArray[0] === checkingArray[1]){
    console.log("match")
    numFlipped = 0;
    checkingArray = [];
    }
  }
}

    var cardDeck = [1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6, 7, 7, 8, 8, 9, 9, 10, 10, 11, 11, 12, 12];

    function shuffle(cardDeck){
      for(var j, x, i = cardDeck.length; i; j = Math.floor(Math.random() * i), x = cardDeck[--i], cardDeck[i] = cardDeck[j], cardDeck[j] = x);
      console.log(cardDeck);
    }
    shuffle(cardDeck);
    for (i = 0; i < cardDeck.length; i++) {
      $('div div').eq(i).val(cardDeck[i]);
    }
    var imageLibrary = {
      1: "images/ermac.png",
      2: "images/jax.png",
      3: "images/johnny_cage.png",
      4: "images/kano.png",
      5: "images/kenshi.png",
      6: "images/liu_kang.png",
      7: "images/quan_chi.png",
      8: "images/radien.png",
      9: "images/reptile.png",
      10: "images/scorpion.png",
      11: "images/sonya_blade.png",
      12: "images/sub_zero.png"
    }

    $("#reset").on("click", function (){
      $('div div').removeClass("flip");
      $("img.frontCard").attr("src", "images/Mortal_Kombat_Logo.png")
    });

    $("#background").on("click", function(){
      $("body").css("background", "url(images/MK_background_coliseum.jpg)");
    });

}); ```
``````

**Comment by andrewsunglaekim** [comment]
if you're storing all of the elements in an array, just programatically remember the last two cards, if they're unmatched, use those stored values to manipulate the dom for just those two elements.

**Comment by pdavies88** [comment]
still struggling on this not exactly sure how to pull the index even though I have it as global also how would I change the DOM to the image I want after I assigned it via .attr("src"). if you have an example that would be great or if we could schedule a time to meet this afternoon that would be helpful

**Comment by andrewsunglaekim** [comment]
I think you could jsut store it in a variable. And yeah i think `.attr()` works really well. I'm in a meeting now, but i can swing by after

**Comment by pdavies88** [comment]
yes that would be great I will try to see if I can get it to work and if I do I'll close the issue if I can't and you can swing by later that would be great. thanks
